src/modules/launcher/Plugins/Wox.Plugin.Program/Programs/Win32.cs

```diff
@@ -462,7 +470,15 @@ public static Win32[] All(Settings settings)
                     programs = programs.Concat(startMenu);
                 }
 
-                return programs.ToArray();
+                var programsWithoutLnk = programs.Where(x => string.IsNullOrEmpty(x.LnkResolvedPath));
+                var programsAsList = programs.ToList();
+
+                Parallel.ForEach(programsWithoutLnk, (app) =>
```

Comment:
I don't think we want Parallel.ForEach here.  We are modifying programsAsList concurrently across multiple threads.  I think this could cause race conditions.